### ISSUE TITLE
Could we get rid of the strict line length check?

### ISSUE THREAD

**Issue by jasperges** [getavalon/core#467]
I completely agree with trying to be PEP8 compliant and using style checkers. But I don't think it's always good to be so strict with the line width. I feel in certain cases it's better to have lines that are a bit longer then 80 characters, because they can greatly improve readability.
See also [this great talk by Raymond Hettinger](https://www.youtube.com/watch?v=wf-BqAjZb8M&t=1s).

See [this code](https://github.com/getavalon/core/blob/master/avalon/tools/workfiles/app.py#L369-L372) from Avalon as an example:
```python
self._messagebox.setStandardButtons(
    self._messagebox.Yes | self._messagebox.No |
    self._messagebox.Cancel
)
```
I quickly broke those lines because `self._messagebox.Yes | self._messagebox.No | self._messagebox.Cancel` was 1 character too long. But now readability suffers IMHO. You could argue of course that it would be even better to firstly assign shorter names to the buttons, so that everything could even fit on 1 line (e.g.: `self._messagebox.setStandardButtons(yes | no | cancel)`, but I hope you get the point.

So in short: although I think it's a very good thing to strive for a maximum line width of 80, sometimes it is better to break this rule in my opinion.

Curious to find out what other people think.

**Comment by tokejepsen** [comment]
Personally I like the strict 80 character rule. I work on a laptop, so being able to half screen code with having scroll sideways is great!

But also what would be the line limit if not 80? No line limit?

> But now readability suffers IMHO.

I dont think it suffers, but I personally have broken all the lines instead of just one.

```python
self._messagebox.setStandardButtons(
    self._messagebox.Yes |
    self._messagebox.No |
    self._messagebox.Cancel
)
```

**Comment by jasperges** [comment]
> But also what would be the line limit if not 80? No line limit?
Most of the time I use 100. But that is of course fairly arbitrary and a personal choice...

> I dont think it suffers, but I personally have broken all the lines instead of just one.
> 
> ```python
> self._messagebox.setStandardButtons(
>     self._messagebox.Yes |
>     self._messagebox.No |
>     self._messagebox.Cancel
> )
> ```
I agree that's better. :)

Let's see if I can find a better example... Now that I think of it, my gripe could very well be with the automatic style checkers and how they 'fix' long lines. A lot of the times, it's better if you do it yourself in my experience.

**Comment by BigRoy** [comment]
In this case I do agree with Toke that if you allow cases to ignore the rule then you might run into discussions about when it is allowed or not, each time. As long as there's a rule that we can consider a decent standard I'm fine with it.

There have been times I struggled with 80 and felt it was too short, and there have also been times I was very happy that code was consistent with the line length... like when having a part of the screen available for the code.

**Comment by tokejepsen** [comment]
> On a related note: how would you feel about including a .pylintrc, .style.yapf, maybe isort.cfg and for python 3 even mypy.ini (maybe even more?) so style checking and formatting is configured for this project.

I'm not familiar with these. What do they do or having for example flake8 linting?

**Comment by tokejepsen** [comment]
Ahh, right, so it works sort of similar to an auto formatter?

**Comment by flaixman** [comment]
autopep8 is another auto formatter that works good too if you want to check it :)

**Comment by jasperges** [comment]
Closing this. The consensus is to stick with a max line width of 80.
